data/district_data/combined_district_keys.csv

```diff
@@ -0,0 +1,91 @@
+Table 1
+PROVINCE;DISTRICT_LEVEL;DATA_TITLE;DATA_TYPE;FIENDLY_TITLE;CAT_B;CAT_LINK;NOTES;DATA_FILE

```

Comment:
> Rename `FIENDLY_TITLE` -> `FRIENDLY_TITLE`.
> 
> Silly question - Is lowercase the general convention in the project for headers? :)

I cannot find any convention for upper and lowercase in this project, in-fact most data files use both. This definition file should set the tone or any future files. I lean towards `UPPERCASE`, but `Sentense` case is my second choice.

Reply:
> Maybe I do not yet fully understanding how this is used by data files just yet. But would we need a ref and use that in the data files or is this used differently in the data files?

1. Standardising this file is the first action for us to take to come to a consensus for all the titles. I am not saying that exiting title should change, therefor the is a `FRIENDLY_NAME` to promote uniform names.

2. Any 3rd party can use this enhanced key file to format their data with the `FRIENDLY_NAME`

3. If everyone agrees, I will create an automated script that will check if the data files comply with these keys. In this way, if fields are broken we will quickly be alerted.

Hope this answers your question?

Reply:
> In terms of structure -I am not sure how far we would want to normalise this data structure. If that is something we would like to do we could consider normalising out `district_level, data_type` and maybe even `data_file` potentially? These could be normalised out into separate key-value files as those seems like dictionaries themselves. But I suppose the difficulty of dealing with normalising CSV files is that you would need to open multiple CSV files to parse a record.

I get that you want to normalise these items. Any one of us would rather want to design a SQL db, but we are stuck with this flat file system. The purpose of the combined key file is to be the single reference key to read.

If we keep using the multiple keys, we are back where we were, as you might lose the standards on the keys again.

Pandas in Python can very easily create multiple indexing with this structure, to work with this file.

Reply:
> Do we need to include `Notes`? Also what is the `CAT_B` and `CAT_LINK` values used for? Sorry for the silly question -I think I might be missing the understanding of what `CAT` in this context mean -I assume some category.

@shaze can motivate that you want to include a wider definition of the keys, especially on the 2nd level.

`CAT_B` (Category B) is the official demarcation code for the area as defined by the Municipal Demarcation Board of South Africa, for 2nd level areas. I only included current keys for WP and GP, to kick us off. The LP data file is completely defined by the demarcation codes. `CAT_LINK` (my own term) links the 2nd level to 1st level areas.

It is a bit confusing at first how they defined their reference table. The first and second level codes are stored in CAT_B, thus I kept this convention.

Even if we did not use the demarcation file before, it enhances the key file very well for 3rd parties. If they have maps with these codes defined. They can easily convert die titles.

Reply:
This is what the daily GDoH report provides. It is also very useful for people since e.g. Tshwane 1 carries no meaning while Ga-Rankuwe	Mabopane Winterveldt	Shoshanguve	Rosslyn	Karenpark	Wonderboom	Akasia	Nina Park carries some meaning. Of course, it is not a complete list (that GDoH) reports -- just a sampling of suburbs (not wards) that make up the district.